xblocks_contrib/html/html.py

```diff
@@ -1,81 +1,134 @@
-                "templates/html.html",
-                {
-                    "count": self.count,
-                },
-                i18n_service=self.runtime.service(self, "i18n"),
+    def get_html(self):
+        """Returns html required for rendering the block."""
+        if self.data:
+            data = self.data
+            user_id = (
+                self.runtime.service(self, "user")
+                .get_current_user()
+                .opt_attrs.get(ATTR_KEY_DEPRECATED_ANONYMOUS_USER_ID)
             )
+            if user_id:
+                data = data.replace("%%USER_ID%%", user_id)
+            data = data.replace("%%COURSE_ID%%", str(self.scope_ids.usage_id.context_key))
+            return data
+        return self.data
+
+    def index_dictionary(self):
+        xblock_body = super().index_dictionary()
+        # Removing script and style
+        html_content = re.sub(
+            re.compile(
+                r"""
+                    <script>.*?</script> |
+                    <style>.*?</style>
+                """,
+                re.DOTALL | re.VERBOSE,
+            ),
+            "",
+            self.data,
         )
-
-        frag.add_css(self.resource_string("static/css/html.css"))
-        frag.add_javascript(self.resource_string("static/js/src/html.js"))
+        html_content = escape_html_characters(html_content)
+        html_body = {
+            "html_content": html_content,
+            "display_name": self.display_name,
+        }
+        if "content" in xblock_body:
+            xblock_body["content"].update(html_body)
+        else:
+            xblock_body["content"] = html_body
+        xblock_body["content_type"] = "Text"
+        return xblock_body
+
+    def student_view_data(self, context=None):  # pylint: disable=unused-argument
+        """
+        Return a JSON representation of the student_view of this XBlock.
+        """
+        if getattr(settings, "FEATURES", {}).get(self.ENABLE_HTML_XBLOCK_STUDENT_VIEW_DATA, False):
+            return {"enabled": True, "html": self.get_html()}
+        else:
+            return {
+                "enabled": False,
+                "message": f'To enable, set FEATURES["{self.ENABLE_HTML_XBLOCK_STUDENT_VIEW_DATA}"]',
+            }
+
+    @XBlock.supports("multi_device")
+    def student_view(self, _context):
+        """
+        Return a fragment that contains the html for the student view
+        """
+        frag = Fragment(self.get_html())
+        frag.add_css(resource_loader.load_unicode("static/css/html.css"))
+        frag.add_javascript("""function HtmlBlock(runtime, element){}""")
         frag.initialize_js("HtmlBlock")
         return frag
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
-    @XBlock.json_handler
-    def increment_count(self, data, suffix=""):
+    def studio_view(self, context):  # pylint: disable=unused-argument
```

Comment:
@kdmccormick @feanil 
Please review this method
What will be the django settings name for this authroing url?

Reply:
Is there a reason to implement this view, rather than simply making this raise NotImplementedError?

By the way, referencing an edx-platform Django setting would make this block dependent on edx-platform, so we should not do that. If you ever need to do that in the future, the right way to do it would be though an XBlock service such as [the settings service](https://github.com/openedx/edx-platform/blob/master/xmodule/services.py#L31)--that way, runtimes other than edx-platform (e.g., xblock-sdk) could implement their own non-Django settings service.